File: domain/src/main/java/com/study/platform/global/idempotency/IdempotencyStoragePort.java
package com.study.platform.global.idempotency;

import java.time.Duration;

public interface IdempotencyStoragePort {

    String get(String key);

    boolean setIfAbsent(String key, String value, Duration ttl);

    void set(String key, String value, Duration ttl);

    void delete(String key);
}

File: domain/src/main/kotlin/com/study/platform/global/idempotency/IdempotencyStoragePort.kt
package com.study.platform.global.idempotency

import java.time.Duration

interface IdempotencyStoragePort {
    fun get(key: String): String?
    fun setIfAbsent(key: String, value: String, ttl: Duration): Boolean
    fun set(key: String, value: String, ttl: Duration)
    fun delete(key: String)
}
